src/components/data/listing/Listing.js

```diff
@@ -30,34 +38,81 @@ function Listing(props) {
     const [rowsPerPage, setRowsPerPage] = useState(25);
     const [loading, setLoading] = useState(false);
+    const [error, setError] = useState("");
     const [data, setData] = useState({ total: 0, files: [], folders: [] });
     const [uploadsCompleted, setUploadsCompleted] = useState(0);
+    const [dataRoots, setDataRoots] = useState([]);
+    const [userHomePath, setUserHomePath] = useState("");
+    const [userTrashPath, setUserTrashPath] = useState("");
 
-    const { baseId, path, handlePathChange } = props;
+    const { baseId, path, handlePathChange, intl } = props;
 
     useEffect(() => {
         setSelected([]);
     }, [path]);
 
+    useEffect(() => {
+        if (path) {
+            callApi({
+                endpoint: `/api/filesystem/paged-directory?path=${path}&limit=${rowsPerPage}&sort-col=${orderBy}&sort-dir=${order}&offset=${rowsPerPage *
+                    page}`,
+                setLoading,
+                setError,
+            }).then((respData) => {
+                respData &&
+                    setData({
+                        total: respData?.total,
+                        listing: [
+                            ...respData?.folders.map((f) => ({
+                                ...f,
+                                type: "FOLDER",
+                            })),
+                            ...respData?.files.map((f) => ({
+                                ...f,
+                                type: "FILE",
+                            })),
+                        ].map((i) => camelcaseit(i)), // camelcase the fields for each object, for consistency.
+                    });
+            });
+        }
+    }, [path, rowsPerPage, orderBy, order, page, uploadsCompleted]);
+
     useEffect(() => {
         callApi({
-            endpoint: `/api/filesystem/paged-directory?path=${path}&limit=${rowsPerPage}&sort-col=${orderBy}&sort-dir=${order}&offset=${rowsPerPage *
-                page}`,
+            endpoint: `/api/filesystem/root`,
             setLoading,
+            setError,
         }).then((respData) => {
-            respData &&
-                setData({
-                    total: respData?.total,
-                    listing: [
-                        ...respData?.folders.map((f) => ({
-                            ...f,
-                            type: "FOLDER",
-                        })),
-                        ...respData?.files.map((f) => ({ ...f, type: "FILE" })),
-                    ].map((i) => camelcaseit(i)), // camelcase the fields for each object, for consistency.
+            if (respData) {
+                const respRoots = respData.roots;
+                const home = respRoots.find(
+                    (root) =>
+                        root.label !== formatMessage(intl, "trash") &&
+                        root.label !== formatMessage(intl, "sharedWithMe") &&
+                        root.label !== formatMessage(intl, "communityData")
```

Comment:
I don't think we need these `formatMessage` bits?  This is being checked against the response we're getting from the service which, I assume, will always be in English.  I may be missing something though.

Reply:
Right. They will in english. May be move them to constants.